### PR TITLE
Include path values in the params passed to the middleware methods

### LINKED ISSUE #1: POST requests don't receive URL placeholder params
In `addItem(params, body, req)` and probably others, the params don't have the URL placeholders, e.g. for `/path/:someId/foo`, the `params` object doesn't have a property for the matching `someId`.

### PR DESCRIPTION
**Pull request by eheikes**
fixes #1

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -85,7 +85,14 @@ Middleware.prototype.getMiddleware = function() {
 		return function(req, res, next) {
 			function parseParams(pathRegExp) {
 				var parsed = pathRegExp.exec(req.url);
+				var mapped = pathRegExp.keys.reduce(function(soFar, key, i) {
```
Comment (eheikes) [inline index.js]: This code adds the keys from the path placeholders to the params.

File: middlewarerule.js [hunk 2]
```diff
@@ -222,8 +222,15 @@ MiddlewareRule.prototype.getCollection = function(params, data, req) {
 	}
 
 	var nonSpecialParams = Object.keys(filteredParams).filter(function(key) {
+		// Exclude special params.
 		return specialParams.indexOf(key) === -1;
-	});
+	}).filter(function(key) {
+		// Exclude path params.
```
Comment (eheikes) [inline middlewarerule.js]: This code excludes the path params from the collection filtering -- only query parameters should be considered.

File: test/test-helpers.js [hunk 3]
```diff
@@ -4,26 +4,29 @@ var middleware = require('../');
 var connect = require('connect');
 var request = require('supertest');
 
-var path = '/foo';
+var path = '/foo/:fooId/bar';
```
Comment (eheikes) [inline test/test-helpers.js]: I modified the default test to have a path placeholder (`fooId`) so it could be tested.

File: middlewarerule.js [hunk 4]
```diff
@@ -222,8 +222,15 @@ MiddlewareRule.prototype.getCollection = function(params, data, req) {
...
+		if (!this.path.keys) { return true; }
+		return this.path.keys.reduce(function(soFar, pathKeyInfo) {
+			return soFar && (pathKeyInfo.name === key);
+		}, true);
```
Comment (jrunning) [inline middlewarerule.js]: Could this `reduce` be replaced with [`Array.prototype.every`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every)?

```js
this.path.keys.every(function(pathKeyInfo) {
  return pathKeyInfo.name === key;
});
```

**Comment by jrunning** [comment]
This LGTM. 👍

**Comment by kddrake** [comment]
👍  code looks good, linted and ran tests successfully

File: middlewarerule.js (hunk 4 above)
Comment (achandrasekar) [inline middlewarerule.js]: Good question @jrunning ! I don't see why not. What do you think @eheikes ?

**Comment by eheikes** [comment]
I replaced the use of `Array.prototype.reduce()` with `every()` or `some()` in 2 places.

**Comment by jrunning** [comment]
:shipit:

**Pull request merged**